network/config/config.ts

```diff
@@ -0,0 +1,93 @@
+import nerfDart from 'nerf-dart';
+
+export interface ParsedUri {
+  /**
+   * The config as string
+   */
+  raw: string;
+  /**
+   * The parsed config
+   */
+  parsed: URL;
+  /**
+   * The nerf dart of the config
+   * @example https://example.com -> //example.com/
+   * @example https://example.com:8080/path/to/file  -> //example.com:8080/path/to/
+   */
+  nerf: string;
+  /**
+   * The host of the config
+   * @example https://example.com -> example.com
+   */
+  host: string;
+  /**
+   * The host of the config with port
+   * @example https://example.com:8080 -> //example.com:8080/
+   */
+  hostOnlyDomain: string;
+  /**
+   * The config without port
+   * @example https://example.com:8080/path/to/file -> https://example.com/path/to/file
+   */
+  withoutPort: string;
+}
+
+export function parseUri(uri: string): ParsedUri {
+  const parsed = new URL(uri);
+
+  if (!uri.endsWith('/')) {
+    uri += '/';
+  }
+
+  return {
+    raw: parsed.href,
+    parsed,
+    nerf: nerfDart(uri),
+    host: parsed.hostname,
+    hostOnlyDomain: convertToDomain(parsed),
+    withoutPort: removePort(parsed),
+  };
+}
```

Comment:
I don't think we need this function because we calculate all these fields but don't always use them, so it is wasting. Like for instance we only used withoutPort after the loop in pickSettingByUrl, so it is enough to call removePort only after the loop